Expand used solhint rules

**🧐 Motivation**

Hey, I was looking into solhint rules and there are ones that may be interesting to use here but I wanted to discuss it before opening an PR


**📝 Details**

'custom-errors',
'explicit-types',
'no-unused-import',

```
contracts/interfaces/IERC1155.sol
  6:9  error  imported name IERC1155 is not used  no-unused-import

contracts/interfaces/IERC1155MetadataURI.sol
  6:9  error  imported name IERC1155MetadataURI is not used  no-unused-import

contracts/interfaces/IERC1155Receiver.sol
  6:9  error  imported name IERC1155Receiver is not used  no-unused-import

contracts/interfaces/IERC165.sol
  6:9  error  imported name IERC165 is not used  no-unused-import

contracts/interfaces/IERC20.sol
  6:9  error  imported name IERC20 is not used  no-unused-import

contracts/interfaces/IERC20Metadata.sol
  6:9  error  imported name IERC20Metadata is not used  no-unused-import

contracts/interfaces/IERC3156.sol
  6:9  error  imported name IERC3156FlashBorrower is not used  no-unused-import
  7:9  error  imported name IERC3156FlashLender is not used    no-unused-import

contracts/interfaces/IERC721.sol
  6:9  error  imported name IERC721 is not used  no-unused-import

contracts/interfaces/IERC721Enumerable.sol
  6:9  error  imported name IERC721Enumerable is not used  no-unused-import

contracts/interfaces/IERC721Metadata.sol
  6:9  error  imported name IERC721Metadata is not used  no-unused-import
```

--- a/package.json
+++ b/package.json
@@ -77,7 +77,7 @@
     "prettier-plugin-solidity": "^1.1.0",
     "rimraf": "^5.0.1",
     "semver": "^7.3.5",
-    "solhint": "^3.3.6",
+    "solhint": "^4.1.1",
     "solhint-plugin-openzeppelin": "file:scripts/solhint-custom",
     "solidity-ast": "^0.4.50",
     "solidity-coverage": "^0.8.5",

--- a/package.json
+++ b/package.json
@@ -77,7 +77,7 @@
     "prettier-plugin-solidity": "^1.1.0",
     "rimraf": "^5.0.1",
     "semver": "^7.3.5",
-    "solhint": "^4.1.1",
+    "solhint": "^3.6.1",
     "solhint-plugin-openzeppelin": "file:scripts/solhint-custom",
     "solidity-ast": "^0.4.50",
     "solidity-coverage": "^0.8.5",

--- a/solhint.config.js
+++ b/solhint.config.js
@@ -1,16 +1,22 @@
 const customRules = require('./scripts/solhint-custom');
 
 const rules = [
-  'no-unused-vars',
+  'avoid-tx-origin',
   'const-name-snakecase',
   'contract-name-camelcase',
   'event-name-camelcase',
+  'explicit-types',
   'func-name-mixedcase',
   'func-param-name-mixedcase',
-  'modifier-name-mixedcase',
-  'var-name-mixedcase',
   'imports-on-top',
+  'modifier-name-mixedcase',
+  'no-console',
   'no-global-import',
+  'no-unused-vars',
+  'quotes',
+  'use-forbidden-name',
+  'var-name-mixedcase',
+  'visibility-modifier-order',
   ...customRules.map(r => `openzeppelin/${r.ruleId}`),
 ];
 

